### PR TITLE
Feature/permissions

### PR DESCRIPTION
**Pull request by c-tn**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,59 +1,87 @@
...
-
-                    <Route path = { routes.profile } component = { Profile }/>
-                    
-                    <Route
-                        path = { routes.logout }
-                        render={ () => <Redirect to = { routes.login } /> }
-                    />
-                </Switch>
+        return (
+            this.state.isLoading
```
Comment (IlyaBobr) [inline client/src/App.js]: You can render header and footer, usually loader shows only for the content section.

File: client/src/App.js [hunk 2]
```diff
@@ -1,59 +1,87 @@
...
         super(props);
 
-        this.checkUserToken();
+        this.state = {
+            isLoading: false
+        }
     }
 
-    checkUserToken() {
+    async componentDidMount() {
```
Comment (IlyaBobr) [inline client/src/App.js]: Why do you use async-await for ComponentDidMount? That's not good idea =)

File: client/src/AppContainer.js [hunk 3]
```diff
@@ -5,7 +5,10 @@ import App from './App'
 import { setAuth } from './Authorization/actions/actionsCreator';
 import { getProfile } from 'Profile/actions/actions';
 
-const mapStateToProps = state => state.auth;
+const mapStateToProps = state => ({
+    auth: state.auth,
+    profile: state.profile
+});
 
 const mapDispatchToProps = dispatch => ({
```
Comment (IlyaBobr) [inline client/src/AppContainer.js]: you also can use plane object if your dispatched function does not have any parameter. 

this ===>
const mapDispatchToProps = dispatch => ({
getProfile : () => dispatch(getProfile())
})
you can transform to this ===> 
const mapDispatchToProps = {
getProfile
}

File: client/src/BookingManagment/actions/actions.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { landings } from 'utils/api';
+
+import { updateLandings } from 'Booking/actions/actionsCreator';
+
+export const createLanding = landingData => dispatch => {
+    landings.createLanding(landingData)
+    .then(({ data }) => {
+        dispatch(updateLandings(data.landings));
+    })
+    .catch(() => {});
```
Comment (IlyaBobr) [inline client/src/BookingManagment/actions/actions.js]: do not use empty catch, just don't use it if you don't need  it

File: client/src/BookingManagment/components/BookingManagment.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import moment from 'moment';
+
+import ManagmentForm from './../containers/BookingManagmentFormContainer';
+import ManagmentList from './../containers/BookingManagmentListContainer';
+
+export default function BookingManagment() {
+    const today = moment().format('DD.MM.Y');
+
+    const [ selectedDate, setSelectedDate ] = useState(today);
```
Comment (IlyaBobr) [inline client/src/BookingManagment/components/BookingManagment.js]: thumbs up for that =) I will really proud if you write your own effect one day =)

File: client/src/BookingManagment/components/BookingManagmentList.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    const deleteLanding = (landing) => {
+        props.deleteLanding({
+            ...landing,
+            date: props.selectedDate
+        });
+    }
+
+    const getAdditionalClasses = (landing) => {
+        if (!props.selectedLanding) return undefined;
```
Comment (IlyaBobr) [inline client/src/BookingManagment/components/BookingManagmentList.js]: it is better to use { } for this

File: client/src/BookingManagment/containers/BookingManagmentFormContainer.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { connect } from 'react-redux';
+
+import BookingManagmentForm from '../components/BookingManagmentForm'
+
+import { changeLanding, createLanding } from './../actions/actions';
+
+const mapStateToProps = state => ({})
```
Comment (IlyaBobr) [inline client/src/BookingManagment/containers/BookingManagmentFormContainer.js]: you can just pass an empty  object or null and don't write mapStateToProps

File: client/src/App.js (hunk 2 above)
Comment (c-tn) [reply client/src/App.js]: > Why do you use async-await for ComponentDidMount? That's not good idea =)

i will rewrite this in a functional style

**Pull request merged**